### ISSUE TITLE
BUG: there is no learning rate parameter for  Conjugate Gradient

### ISSUE THREAD

**Issue by Sandy4321** [scipy/scipy#15511]
### Describe your issue.

there is no learning rate parameter for  Conjugate Gradient
https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.linalg.cg.html

per paper 
https://arxiv.org/abs/1009.5839

learning rate should be specified 

### Reproducing Code Example

```python
no code example 
no need
```

### Error message

```shell
no error message 
no need
```

### SciPy/NumPy/Python version information

latest

**Comment by rkern** [comment]
The "conjugate gradient" algorithm referred to in the paper (optimizing a kernel regression model) is not the conjugate gradient used in `scipy.sparse.linalg.cg()` (solving a linear system). Both are nominally related to a very general "conjugate gradient" technique for solving general continuous optimization problems, their specific formulations for their specific optimization problems don't necessarily have the same input requirements. Solving a linear system is a _very_ constrained problem that permits the step size to be automatically determined and does not require a learning rate parameter to scale the step size. [An Introduction to the Conjugate Gradient Method Without the Agonizing Pain](https://www.cs.cmu.edu/~quake-papers/painless-conjugate-gradient.pdf) is a classic reference.

**Comment by Sandy4321** [comment]
great thanks for soon answer
"Solving a linear system is a very constrained problem that permits the step size to be automatically determined"

thanks for explaining, sorry for false alarm

it would be very kind of you to help to find plain python code from scratch to better understand
 
 how conjugate gradient works without learning rate tuning 
especially in ridge regression case for 
https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.Ridge.html

my guess scikit-learn Ridge regression  uses  your conjugate gradient code implementation

**Comment by Sandy4321** [comment]
great code was found
https://towardsdatascience.com/complete-step-by-step-conjugate-gradient-algorithm-from-scratch-202c07fb52a8

but still may you help to understand how your implementation deals with non-symmetric matrices
as mentioned in 
https://en.wikipedia.org/wiki/Conjugate_gradient_method
https://en.wikipedia.org/wiki/Biconjugate_gradient_method
https://en.wikipedia.org/wiki/Biconjugate_gradient_stabilized_method

**Comment by Sandy4321** [comment]
like
https://stats.hohoweiya.xyz/2017/08/13/Conjugate-Gradient-for-Regression/
![image](https://user-images.githubusercontent.com/11426119/152821197-1a76679c-c38e-494b-afa4-d1519a42ca75.png)

**Comment by rkern** [comment]
I'm afraid this isn't going to be a forum where you will get this kind of general help. But to clarify a couple of relevant points:

1. `scipy.sparse.linalg.cg()` _does_ only handle symmetric positive-definite matrices.
2. `RidgeRegression` constructs a symmetric positive-definite operator to pass to `cg()` (see `create_mv()` for how it is constructed).

**Comment by Sandy4321** [comment]
thanks for soon valuable answer
